mlrun/api/utils/clients/iguazio.py

```diff
@@ -65,7 +65,9 @@ def __init__(self) -> None:
         self._session = requests.Session()
         self._session.mount("http://", http_adapter)
         self._api_url = mlrun.mlconf.iguazio_api_url
-        self._wait_for_job_completion_retry_interval = 5
+        self._wait_for_job_completion_retry_interval = mlrun.utils.create_step_backoff(
```

Comment:
* the numbers here are optimized for project creation/deletion jobs, but will be used to any call for wait for job completion, that should be changed.
* The `[2, 1]` is very specific and counting on your knowledge that iguazio's internal waits are 1 second and there are 2 jobs, that might change, and no one will come here to update it, I suggest to simply go with `[[1, 10], [5, None]]`
* Generally, if you are putting so specific numbers in code, you have to add a comment explaining the rationale behind them